lib/benchee/formatters/json.ex

```diff
@@ -59,43 +57,44 @@ defmodule Benchee.Formatters.JSON do
       ...>            percentiles:   %{99 => 900}
       ...>          }
       ...>        },
-      ...>      ],
-      ...>      configuration: %Benchee.Configuration{
-      ...>        formatter_options: %{json: %{file: "my_file.json"}}
-      ...>      }
+      ...>      ]
       ...>    }
-      iex> Benchee.Formatters.JSON.format(suite)
-      {%{"Some Input" => "{\\"run_times\\":{\\"My Job\\":[200,400,400,400,500,500,700,900]},\\"sort_order\\":[\\"My Job\\"],\\"statistics\\":{\\"My Job\\":{\\"average\\":500.0,\\"ips\\":2.0e3,\\"maximum\\":900,\\"median\\":450.0,\\"minimum\\":200,\\"mode\\":400,\\"percentiles\\":{\\"99\\":900},\\"sample_size\\":8,\\"std_dev\\":200.0,\\"std_dev_ips\\":800.0,\\"std_dev_ratio\\":0.4}}}"}, "my_file.json"}
+      iex> Benchee.Formatters.JSON.format(suite, %{file: "my_file.json"})
+      %{"Some Input" => "{\\"run_times\\":{\\"My Job\\":[200,400,400,400,500,500,700,900]},\\"sort_order\\":[\\"My Job\\"],\\"statistics\\":{\\"My Job\\":{\\"average\\":500.0,\\"ips\\":2.0e3,\\"maximum\\":900,\\"median\\":450.0,\\"minimum\\":200,\\"mode\\":400,\\"percentiles\\":{\\"99\\":900},\\"sample_size\\":8,\\"std_dev\\":200.0,\\"std_dev_ips\\":800.0,\\"std_dev_ratio\\":0.4}}}"}
 
   """
-  @spec format(Suite.t) :: {%{Suite.key => String.t}, String.t}
-  def format(%Suite{scenarios: scenarios, configuration:
-               %Configuration{formatter_options: %{json: %{file: filename}}}}) do
-    data = scenarios
-           |> Enum.group_by(fn(scenario) -> scenario.input_name end)
-           |> Enum.map(fn({input, scenarios}) ->
-               {input, format_scenarios_for_input(scenarios)}
-             end)
-           |> Map.new
-    {data, filename}
+  @spec format(Suite.t(), map) :: %{Suite.key() => String.t()}
+  def format(%Suite{scenarios: scenarios}, %{file: _}) do
+    scenarios
+    |> Enum.group_by(fn scenario -> scenario.input_name end)
+    |> Enum.map(fn {input, scenarios} ->
+      {input, format_scenarios_for_input(scenarios)}
+    end)
+    |> Map.new()
```

Comment:
Love how this transforms into a full pipe as a side effect of the change :muscle: